qrscanner/src/main/java/io/gnosis/safe/qrscanner/QRCodeDecoder.kt

```diff
@@ -0,0 +1,33 @@
+package io.gnosis.safe.qrscanner
+
+import com.google.zxing.BinaryBitmap
+import com.google.zxing.LuminanceSource
+import com.google.zxing.PlanarYUVLuminanceSource
+import com.google.zxing.common.HybridBinarizer
+import com.google.zxing.qrcode.QRCodeReader
+
+/*
+ * Check https://github.com/walleth/walleth/tree/master/app/src/main/java/org/walleth/activities/qrscan
+ */
+class QRCodeDecoder {
+    private val reader = QRCodeReader()
+
+    private fun LuminanceSource.decode() =
+        reader.decode(BinaryBitmap(HybridBinarizer(this))).text
+
+    fun decode(data: ByteArray, width: Int, height: Int): String {
+        val centerX = width / 2
+        val centerY = height / 2
+
+        var size = width.coerceAtMost(height)
+        size = (size.toDouble() * ReticleView.FRAME_SCALE).toInt()
+
+        val halfSize = size / 2
+
+        val left = centerX - halfSize
+        val top = centerY - halfSize
+
+        val source = PlanarYUVLuminanceSource(data, width, height, left, top, size, size, false)
+        return nullOnThrow { source.decode() } ?: source.invert().decode()
```

Comment:
Maybe we could a comment to this line. So we catch it once and invert the image, because some websites use white QR codes on black background which is not supported by the default `.decode`. This was the workaround that worked the best for me. If anyone wants to thest this they can use uniswap in dark mode (there they use an inverted QR code)